### ISSUE TITLE
MinGW build fails on linking

### ISSUE THREAD

**Issue by wwylele** [citra-emu/citra#2821]
After #2803, which adds a new dependency enet, MinGW errors the following in the final linking stage:
```
../../externals/enet/libenet.a(win32.c.obj):win32.c:(.text+0x38): undefined reference to `__imp_timeBeginPeriod'
../../externals/enet/libenet.a(win32.c.obj):win32.c:(.text+0x8b): undefined reference to `__imp_timeEndPeriod'
../../externals/enet/libenet.a(win32.c.obj):win32.c:(.text+0xa3): undefined reference to `__imp_timeGetTime'
../../externals/enet/libenet.a(win32.c.obj):win32.c:(.text+0xb6): undefined reference to `__imp_timeGetTime'
../../externals/enet/libenet.a(win32.c.obj):win32.c:(.text+0xd9): undefined reference to `__imp_timeGetTime'
collect2.exe: error: ld returned 1 exit status
src/citra_qt/CMakeFiles/citra-qt.dir/build.make:830: recipe for target 'src/citra_qt/citra-qt.exe' failed
mingw32-make[2]: *** [src/citra_qt/citra-qt.exe] Error 1
CMakeFiles/Makefile2:1082: recipe for target 'src/citra_qt/CMakeFiles/citra-qt.dir/all' failed
mingw32-make[1]: *** [src/citra_qt/CMakeFiles/citra-qt.dir/all] Error 2
Makefile:94: recipe for target 'all' failed
mingw32-make: *** [all] Error 2
```

These functions are from winmm library. The following hack is confirmed working:
add `target_link_libraries(enet PRIVATE winmm ws2_32)` to enet's `CMakeLists.txt` (ws2_32 is also needed, otherwise similar linking error will be generated)

What doesn't make sense to me is that [we already added winmm and ws2_32 to `PLATFORM_LIBRARIES` in the root `CMakeLists.txt`](https://github.com/citra-emu/citra/blob/master/CMakeLists.txt#L241), so I don't know what happens there...

BTW, can we add MinGW build to the main repo CI? cc @jroweboy

**Comment by Anakonda** [comment]
Mingw clearly states that per GCC manual libriaries must be specified after the names of all dependent modules. So ws2_32 should be about last library you add.
see -l in http://gcc.gnu.org/onlinedocs/gcc/Link-Options.html#Link-Options

**Comment by wwylele** [comment]
@Anakonda I think that weird stuff is handled by CMake? Plus our Linux GCC build still works fine

**Comment by Anakonda** [comment]
Mingw wiki says some GCC on linux ignore this, and ws2_32 is Windows socket api and windows only
Also CMake adds -l in the order of target_link_libraries call.

**Comment by wwylele** [comment]
@Anakonda anyway, they are almost added as the last library: https://github.com/citra-emu/citra/blob/master/src/citra_qt/CMakeLists.txt#L96. `LATFORM_LIBRARIES` includes the two libraries in question.

**Comment by jroweboy** [comment]
@wwylele The issue is what @Anakonda was alluding to. `enet` needs to link against `winmm` on mingw, but does not include that in the `CMakeLists.txt`. Because of this, cmake does not add the required -lwinmm and -lws2_32 after the static library as is required. Your work around fixes this because it adds them in the call. Maybe a simple way to explain it would be to show the full compile command :)

Before "hack"
```bash
/C/msys64/mingw64/bin/g++.exe {scrubbed} -lwinmm -lws2_32 {scrubbed} ../../externals/enet/libenet.a /C/msys64/mingw64/lib/libQt5Gui.dll.a /C/msys64/mingw64/lib/libQt5Core.dll.a -lws2_32 -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -luuid -lcomdlg32 -ladvapi32
```

After "hack"
```bash
/C/msys64/mingw64/bin/g++.exe {scrubbed} -lwinmm -lws2_32 {scrubbed} ../../externals/enet/libenet.a -lwinmm -lws2_32 /C/msys64/mingw64/lib/libQt5Gui.dll.a /C/msys64/mingw64/lib/libQt5Core.dll.a -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -luuid -lcomdlg32 -ladvapi32
```

Notice that it explicitly links `winmm` and `ws2_32` immediately after libenet.a which allows the linker to find the missing functions.

Theres probably some other solutions out there, but maybe this is best to try to upstream it? Something else we can do is just add that line into the externals cmake file. Should be just as good, but i'd like to at least try to upstream the fix.

**Comment by wwylele** [comment]
fixed by #2824
